config/src/main/java/org/springframework/security/config/annotation/web/configuration/ConversionServicePostProcessor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.config.annotation.web.configuration;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.convert.converter.ConverterRegistry;
+import org.springframework.core.convert.support.ConfigurableConversionService;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.security.crypto.factory.RsaKeyConverters;
+
+/**
+ * Adds Spring Security converters to the configured {@link ConversionService}
+ *
+ * @author Josh Cummings
+ * @since 5.2
+ */
+public class ConversionServicePostProcessor implements BeanFactoryPostProcessor {
```

Comment:
I wonder if there is a reason this is so generic given it adds RSA converts. I also wonder if it makes sense in the current package since RSA is applicable to non web apps and to reactive applications.

Reply:
A quick search across the code base shows a handful of classes that extend `BeanFactoryPostProcessor`. For example, there is one for `ldap`, one for `debug`, and one for `websocket`, among others.

Each of these is in `spring-security-config` and follows the pattern `org.springframework.security.config.{module}`.

So, it seems natural to rename this as `org.springframework.security.config.crypto.RsaKeyConversionServicePostProcessor` since it is post processing the `ConversionService` to support RSA keys.